### ISSUE TITLE
[BUG] Missing CPU fallback for GetMapValue on scalar map, vector key

### ISSUE THREAD

**Issue by mythrocks** [NVIDIA/spark-rapids#6282]
When looking up a `scalar` `map` using a vector of `keys`, the `spark-plugin` fails to fall back to CPU.

Repro case:
```scala
import org.apache.spark.sql._
import org.apache.spark.sql.types._

val schema = StructType( Array( StructField("key", IntegerType) ) )
val rows = Array( Row(1), Row(2) )

spark.createDataFrame(sc.parallelize(rows), schema).createOrReplaceTempView("mytable")

sql("SELECT key, MAP(1, 'one', 2, 'two')[key] FROM mytable  ").show
```
The query fails with the following exception:
```
22/08/10 17:38:09 ERROR Executor: Exception in task 2.0 in stage 1.0 (TID 3)java.lang.IllegalStateException: Map lookup keys must be scalar values        at org.apache.spark.sql.rapids.GpuGetMapValue.doColumnar(complexTypeExtractors.scala:231)
        at com.nvidia.spark.rapids.GpuBinaryExpression.$anonfun$columnarEval$3(GpuExpressions.scala:258) 
        at com.nvidia.spark.rapids.Arm.withResourceIfAllowed(Arm.scala:73)
        at com.nvidia.spark.rapids.Arm.withResourceIfAllowed$(Arm.scala:71)
        at org.apache.spark.sql.rapids.GpuGetMapValue.withResourceIfAllowed(complexTypeExtractors.scala:190)
        at com.nvidia.spark.rapids.GpuBinaryExpression.$anonfun$columnarEval$2(GpuExpressions.scala:253) 
        at com.nvidia.spark.rapids.Arm.withResourceIfAllowed(Arm.scala:73)        at com.nvidia.spark.rapids.Arm.withResourceIfAllowed$(Arm.scala:71)
...
```

Ideally, this query should have succeeded by falling back from `GpuGetMapValue` to `GetMapValue`.

**Comment by mythrocks** [comment]
For `22.08`, the `spark-plugin` should probably just fall back to CPU, since `cudf::jni::maps_column_view` does not support vector keys on scalar maps. I'm working on this now.

In `22.10`, we probably need a `cudf::jni::maps_scalar_view`, to support key vector lookup.

**Comment by mythrocks** [comment]
Fixed by #6288. Verified to fall back to CPU. 
We'll add actual support for acceleration when #6296 is addressed.
